semantic/lookup.go

```diff
@@ -0,0 +1,99 @@
+package semantic
+
+import (
+	flatbuffers "github.com/google/flatbuffers/go"
+	"github.com/influxdata/flux/codes"
+	"github.com/influxdata/flux/internal/errors"
+	"github.com/influxdata/flux/libflux/go/libflux"
+	"github.com/influxdata/flux/semantic/internal/fbsemantic"
+)
+
+// LookupBuiltInType returns the type of the builtin value for a given
+// Flux stdlib package. Returns a empty MonoType if lookup fails.
+func LookupBuiltInType(pkg, name string) (MonoType, error) {
+	byteArr := libflux.EnvStdlib()
+	env := fbsemantic.GetRootAsTypeEnvironment(byteArr, 0)
+	var table flatbuffers.Table
+	// perform a lookup of package identifier in flatbuffers TypeEnvironment
+	prop, err := lookup(env, pkg, name)
+	if err != nil {
+		return MonoType{}, err
+	}
+	if !prop.V(&table) {
+		return MonoType{}, errors.Newf(codes.Internal, "Prop value is not valid: %v", err)
+	}
+	monotype, err := NewMonoType(table, prop.VType())
+	if err != nil {
+		return MonoType{}, err
+	}
+	// return fb polytype within semantic wrapper
+	return monotype, nil
+}
+
+// lookup is a helper function that performs a lookup of a package identifier in
+// a flatbuffers TypeEnvironment. It first checks for the package using path string
+// and then checks for the package identifier, the "name" string in this case, returning
+// the corresponding MonoType if found
+func lookup(env *fbsemantic.TypeEnvironment, pkg, name string) (*fbsemantic.Prop, error) {
+	// Find package
+	typeAssign := new(fbsemantic.TypeAssignment)
+	if pkgErr := foundPackage(env, typeAssign, pkg); pkgErr != nil {
+		return nil, pkgErr
+	}
+
+	// Grab PolyType expr; expr should always be of type MonoTypeRow
+	polytype := typeAssign.Ty(nil)
+	if polytype.ExprType() != fbsemantic.MonoTypeRow {
+		return nil, errors.Newf(
+			codes.Internal,
+			"Expected PolyType expr to be fbsemantic.MonoTypeRow; found %v",
+			polytype.ExprType(),
+		)
+	}
+	var table flatbuffers.Table
+	if !polytype.Expr(&table) {
+		return nil, errors.New(codes.Internal, "PolyType expr is not valid")
+	}
+
+	// check for package identifier in Row Props
+	row := new(fbsemantic.Row)
+	row.Init(table.Bytes, table.Pos)
+	prop := new(fbsemantic.Prop)
+	if propErr := foundProp(row, prop, name); propErr != nil {
+		return nil, propErr
+	}
+
+	return prop, nil
+}
+
+// foundPackage is a helper function that iterates over type assignments and checks
+// for a given package returning an error if the package is not found
+func foundPackage(env *fbsemantic.TypeEnvironment, obj *fbsemantic.TypeAssignment, pkg string) error {
```

Comment:
Just a style comment here: it would be more idiomatic and easier to read if you created a `*fbsemantic.TypeAssignment` in this method, and returned it as a value, rather than have the caller create it and pass it as a pointer.  So the signature to this method would be something like 
```Go
func foundPackage(env *fbsemantic.TypeEnvironment, pkg string) (*fbsemantic.TypeAssignment, error)
```

Another, nit-like comment, it's good when method names contain verbs, but they're typically in the present tense, so this method might be better named `findPackage`.

Reply:
Yeah, these functions might go away after I merge #2377 anyway, but those are good notes. If I keep either func, I'll add these changes into that PR.